File: src/main/java/dev/sky_lock/pocketlifevehicle/Permission.java
package dev.sky_lock.pocketlifevehicle;

import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

/**
 * @author sky_lock
 */

public enum Permission {
    ADMIN_COMMAND("plvehicle.command.admin.use"),
    CAR_CLICK("plvehicle.entity.open-gui"),
    CAR_PLACE("plvehicle.entity.place");

    private final String permission;
    Permission(String permission) {
        this.permission = permission;
    }

    public boolean obtained(Player player) {
        return player.hasPermission(permission);
    }

    public boolean obtained(CommandSender sender) {
        return sender.hasPermission(permission);
    }
}

File: src/main/kotlin/dev/sky_lock/pocketlifevehicle/Permission.kt
package dev.sky_lock.pocketlifevehicle

import org.bukkit.permissions.Permissible

/**
 * @author sky_lock
 */

 enum class Permission(private val permission: String) {
    ADMIN_COMMAND("plvehicle.command.admin.use"),
    OPEN_VEHICLE_GUI("plvehicle.entity.gui.open"),
    OPEN_EVENT_VEHICLE_GUI("plvehicle.entity.event.gui.open"),
    PLACE_OTHER_VEHICLE("plvehicle.entity.place.other");

    fun obtained(permissible: Permissible): Boolean {
        return permissible.hasPermission(permission)
    }
}